src/components/common/Dialogs/ControlSafeDialog/ControlSafeForm.tsx

```diff
@@ -0,0 +1,281 @@
+    defaultMessage: `Control Safe`,
+  },
+  description: {
+    id: `${displayName}.description`,
+    defaultMessage: `You can use Control Safe to interact with other third party smart contracts. Be careful. <a>Learn more</a>`,
+  },
+  selectSafe: {
+    id: `${displayName}.selectSafe`,
+    defaultMessage: `Select Safe`,
+  },
+  safePickerPlaceholder: {
+    id: `${displayName}.safePickerPlaceholder`,
+    defaultMessage: `Select Safe to control`,
+  },
+  transactionLabel: {
+    id: `${displayName}.transactionLabel`,
+    defaultMessage: `Select transaction type`,
+  },
+  transactionPlaceholder: {
+    id: `${displayName}.transactionPlaceholder`,
+    defaultMessage: `Select transaction`,
+  },
+  buttonTransaction: {
+    id: `${displayName}.buttonTransaction`,
+    defaultMessage: `Add another transaction`,
+  },
+  invalidSafeError: {
+    id: `${displayName}.invalidSafeError`,
+    defaultMessage: `Select a safe from the menu or add a new one via Safe Control`,
+  },
+  warningIconTitle: {
+    id: `${displayName}.warningIconTitle`,
+    defaultMessage: `Warning!`,
+  },
+  upgradeWarning: {
+    id: `${displayName}.upgradeWarning`,
+    defaultMessage: `Controlling a Safe is not supported on your current Colony version. Please upgrade Colony to at least version 12.{break}You can do this via <span>New Action > Advanced > Upgrade</span>`,
+  },
+});
+
+export const { invalidSafeError } = MSG;
+
+const ReadMoreLink = (chunks: React.ReactNode[]) => (
+  <ExternalLink href={SAFE_INTEGRATION_LEARN_MORE}>{chunks}</ExternalLink>
+);
+
+const UpgradeWarning = (chunks: React.ReactNode[]) => (
+  <span className={styles.upgradePath}>{chunks}</span>
+);
+
+const requiredRoles: ColonyRole[] = [ColonyRole.Root];
+
+const hardcodedSafes: Safe[] = [
+  {
+    name: 'test',
+    chainId: 56,
+    address: '0xe759f388c97674D5B8a60406b254aB59f194163d',
+    moduleContractAddress: '0xe759f388c97674D5B8a60406b254aB59f194163d',
+  },
+  {
+    name: 'test2',
+    chainId: 1,
+    address: '0x3F107AFF0342Cfb5519A68B3241565F6FC9BAC1e',
+    moduleContractAddress: '0xe759f388c97674D5B8a60406b254aB59f194163w',
+  },
+  {
+    name: 'test3',
+    chainId: 5,
+    address: '0x3F107AFF0342Cfb5519A68B3241565F6FC9BAC2e',
+    moduleContractAddress: '0xe759f388c97674D5B8a60406b254aB59f194163e',
+  },
+];
+
+const ControlSafeForm = ({
+  back,
+  colony,
+  colony: { version, metadata },
+  enabledExtensionData,
+}: ControlSafeProps) => {
+  const [prevSafeAddress, setPrevSafeAddress] = useState<string>('');
+
+  const {
+    formState: { isSubmitting, dirtyFields },
+    watch,
+  } = useFormContext();
+  const selectedSafe = watch('safe');
+  const transactionType = watch('transactions.transactionType');
+
+  const { userHasPermission } = useActionDialogStatus(
+    colony,
+    requiredRoles,
+    [Id.RootDomain],
+    enabledExtensionData,
+  );
+
+  const submitButtonText = (() => {
+    return { id: 'button.confirm' };
+  })();
+
```

Comment:
why we wrote this as a function and call this function immediately?

Reply:
there is a bit more logic that needs to be added later that changes this text so this is a placeholder until we get around to adding that extra logic back in